{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6eedade0-3f44-4e22-a483-e5ff9d5e819c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import libraries\n",
    "import pandas as pd\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "# disable warnings, use w caution\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "# project specific libs\n",
    "import os\n",
    "import labtools.statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d78b2c48-4fa5-4a54-95b6-00e334849a3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# set path\n",
    "path = '/Users/KevinBu/Desktop/clemente_lab/Projects/ampaim/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49cf0bbf-bc4c-4ffe-ba61-6db8841a46ff",
   "metadata": {},
   "source": [
    "##### Metadata #####"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c30a4d2a-3f3d-4e86-854b-9c6f0374f789",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Diagnosis\n",
       "PsA        144\n",
       "SjD         85\n",
       "RA          72\n",
       "PsO         53\n",
       "SLE         22\n",
       "IBD          4\n",
       "Control      3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in data\n",
    "df = pd.read_csv(path + 'inputs/samples.txt', sep='\\t',index_col=0)\n",
    "\n",
    "# trim sample ids and drop dups\n",
    "df.index = df.index.map(lambda x: x.split('_S')[0])\n",
    "df = df[~df.index.duplicated(keep='first')]\n",
    "\n",
    "# rename index and drop q2 shenanigans\n",
    "df.index.name = 'SampleID'\n",
    "df=df.drop('#q2:types')\n",
    "\n",
    "# autofill diagnoses\n",
    "def get_dx(x):\n",
    "    if 'psa' in x:\n",
    "        return 'PsA'\n",
    "    elif 'ra' in x:\n",
    "        return 'RA'\n",
    "    elif 'pso' in x:\n",
    "        return 'PsO'\n",
    "    elif 'lockit' in x:\n",
    "        return 'SLE'\n",
    "    elif 'ellipss' in x:\n",
    "        return 'PsO'\n",
    "    elif 'sle' in x:\n",
    "        return 'SLE'\n",
    "    elif 'stamp' in x:\n",
    "        return 'SjD'\n",
    "    elif 'ibd' in x:\n",
    "        return 'IBD'\n",
    "    elif 'MOC':\n",
    "        return 'Control'\n",
    "    else:\n",
    "        return 'NA'\n",
    "\n",
    "df['Diagnosis'] = df.index.map(lambda x: get_dx(x))\n",
    "df.to_csv(path + 'inputs/ampaim_metadata.txt', sep='\\t')\n",
    "df.Diagnosis.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e765e98b-51c9-4504-89f1-1f99e1866cba",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Subject_ID</th>\n",
       "      <th>Include in AMP AIM analyses?</th>\n",
       "      <th>AMP AIM</th>\n",
       "      <th>Project</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>Plate</th>\n",
       "      <th>Number_Sample_Plate</th>\n",
       "      <th>Template_DNA_well</th>\n",
       "      <th>Amplicon_Well</th>\n",
       "      <th>Amp_Well_Plate</th>\n",
       "      <th>...</th>\n",
       "      <th>Seq_Barcode</th>\n",
       "      <th>Plates_For_March_Sequencing_2014</th>\n",
       "      <th>BARCODE_PLATE_GTC</th>\n",
       "      <th>Loc_Barcode_Plate_GTC</th>\n",
       "      <th>Barcode_Label_GTC</th>\n",
       "      <th>Seq_Barcode_GTC</th>\n",
       "      <th>Unnamed: 44</th>\n",
       "      <th>Unnamed: 45</th>\n",
       "      <th>MiSeq_Folder</th>\n",
       "      <th>Notes</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NEG-4</th>\n",
       "      <td>NEG.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>neg</td>\n",
       "      <td>neg</td>\n",
       "      <td>348</td>\n",
       "      <td>75</td>\n",
       "      <td>C10</td>\n",
       "      <td>C10</td>\n",
       "      <td>348_75</td>\n",
       "      <td>...</td>\n",
       "      <td>515rcbc513</td>\n",
       "      <td>TCCTAGGTCCGA</td>\n",
       "      <td>6</td>\n",
       "      <td>C10</td>\n",
       "      <td>515rcbc513</td>\n",
       "      <td>TCCTAGGTCCGA</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>MSQ152</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>https://genome.med.nyu.edu/results/segallab/20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MOC-348</th>\n",
       "      <td>MOC.348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>moc</td>\n",
       "      <td>moc</td>\n",
       "      <td>348</td>\n",
       "      <td>76</td>\n",
       "      <td>D10</td>\n",
       "      <td>D10</td>\n",
       "      <td>348_76</td>\n",
       "      <td>...</td>\n",
       "      <td>515rcbc525</td>\n",
       "      <td>CTCTTCTGATCA</td>\n",
       "      <td>6</td>\n",
       "      <td>D10</td>\n",
       "      <td>515rcbc525</td>\n",
       "      <td>CTCTTCTGATCA</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>MSQ152</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>https://genome.med.nyu.edu/results/segallab/20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209-pso</th>\n",
       "      <td>209.pso</td>\n",
       "      <td>YES</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atrisk</td>\n",
       "      <td>pso</td>\n",
       "      <td>349</td>\n",
       "      <td>41</td>\n",
       "      <td>A6</td>\n",
       "      <td>A6</td>\n",
       "      <td>349_41</td>\n",
       "      <td>...</td>\n",
       "      <td>515rcbc485</td>\n",
       "      <td>GAGAGGGATCAC</td>\n",
       "      <td>6</td>\n",
       "      <td>A6</td>\n",
       "      <td>515rcbc485</td>\n",
       "      <td>GAGAGGGATCAC</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>MSQ152</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>https://genome.med.nyu.edu/results/segallab/20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241-pso</th>\n",
       "      <td>241.pso</td>\n",
       "      <td>YES</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atrisk</td>\n",
       "      <td>pso</td>\n",
       "      <td>349</td>\n",
       "      <td>42</td>\n",
       "      <td>B6</td>\n",
       "      <td>B6</td>\n",
       "      <td>349_42</td>\n",
       "      <td>...</td>\n",
       "      <td>515rcbc497</td>\n",
       "      <td>TTGCGACAAAGT</td>\n",
       "      <td>6</td>\n",
       "      <td>B6</td>\n",
       "      <td>515rcbc497</td>\n",
       "      <td>TTGCGACAAAGT</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>MSQ152</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>https://genome.med.nyu.edu/results/segallab/20...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260-pso</th>\n",
       "      <td>260.pso</td>\n",
       "      <td>YES</td>\n",
       "      <td>NaN</td>\n",
       "      <td>atrisk</td>\n",
       "      <td>pso</td>\n",
       "      <td>349</td>\n",
       "      <td>43</td>\n",
       "      <td>C6</td>\n",
       "      <td>C6</td>\n",
       "      <td>349_43</td>\n",
       "      <td>...</td>\n",
       "      <td>515rcbc509</td>\n",
       "      <td>CAGTGTCATGAA</td>\n",
       "      <td>6</td>\n",
       "      <td>C6</td>\n",
       "      <td>515rcbc509</td>\n",
       "      <td>CAGTGTCATGAA</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>MSQ152</td>\n",
       "      <td>1/23/25</td>\n",
       "      <td>https://genome.med.nyu.edu/results/segallab/20...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Subject_ID Include in AMP AIM analyses? AMP AIM Project Diagnosis  \\\n",
       "SampleID                                                                     \n",
       "NEG-4         NEG.4                          NaN     NaN     neg       neg   \n",
       "MOC-348     MOC.348                          NaN     NaN     moc       moc   \n",
       "209-pso     209.pso                          YES     NaN  atrisk       pso   \n",
       "241-pso     241.pso                          YES     NaN  atrisk       pso   \n",
       "260-pso     260.pso                          YES     NaN  atrisk       pso   \n",
       "\n",
       "          Plate  Number_Sample_Plate Template_DNA_well Amplicon_Well  \\\n",
       "SampleID                                                               \n",
       "NEG-4       348                   75               C10           C10   \n",
       "MOC-348     348                   76               D10           D10   \n",
       "209-pso     349                   41                A6            A6   \n",
       "241-pso     349                   42                B6            B6   \n",
       "260-pso     349                   43                C6            C6   \n",
       "\n",
       "         Amp_Well_Plate  ... Seq_Barcode Plates_For_March_Sequencing_2014  \\\n",
       "SampleID                 ...                                                \n",
       "NEG-4            348_75  ...  515rcbc513                     TCCTAGGTCCGA   \n",
       "MOC-348          348_76  ...  515rcbc525                     CTCTTCTGATCA   \n",
       "209-pso          349_41  ...  515rcbc485                     GAGAGGGATCAC   \n",
       "241-pso          349_42  ...  515rcbc497                     TTGCGACAAAGT   \n",
       "260-pso          349_43  ...  515rcbc509                     CAGTGTCATGAA   \n",
       "\n",
       "         BARCODE_PLATE_GTC Loc_Barcode_Plate_GTC  Barcode_Label_GTC  \\\n",
       "SampleID                                                              \n",
       "NEG-4                    6                   C10         515rcbc513   \n",
       "MOC-348                  6                   D10         515rcbc525   \n",
       "209-pso                  6                    A6         515rcbc485   \n",
       "241-pso                  6                    B6         515rcbc497   \n",
       "260-pso                  6                    C6         515rcbc509   \n",
       "\n",
       "          Seq_Barcode_GTC  Unnamed: 44  Unnamed: 45 MiSeq_Folder  \\\n",
       "SampleID                                                           \n",
       "NEG-4        TCCTAGGTCCGA      1/23/25       MSQ152      1/23/25   \n",
       "MOC-348      CTCTTCTGATCA      1/23/25       MSQ152      1/23/25   \n",
       "209-pso      GAGAGGGATCAC      1/23/25       MSQ152      1/23/25   \n",
       "241-pso      TTGCGACAAAGT      1/23/25       MSQ152      1/23/25   \n",
       "260-pso      CAGTGTCATGAA      1/23/25       MSQ152      1/23/25   \n",
       "\n",
       "                                                      Notes  \n",
       "SampleID                                                     \n",
       "NEG-4     https://genome.med.nyu.edu/results/segallab/20...  \n",
       "MOC-348   https://genome.med.nyu.edu/results/segallab/20...  \n",
       "209-pso   https://genome.med.nyu.edu/results/segallab/20...  \n",
       "241-pso   https://genome.med.nyu.edu/results/segallab/20...  \n",
       "260-pso   https://genome.med.nyu.edu/results/segallab/20...  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# RB md\n",
    "df_rb = pd.read_csv(path + 'inputs/Mapping file_yonghua_021925v5_rbb.txt', sep='\\t',index_col=0)\n",
    "df_rb.index.name = 'SampleID'\n",
    "df_rb = df_rb.dropna(subset='Subject_ID')\n",
    "\n",
    "# drop cols of all na\n",
    "df_rb = df_rb.dropna(how='all',axis=1)\n",
    "\n",
    "# drop halo; 356 samples all seem to be accounted for :) in Adam's downloads\n",
    "df_rb = df_rb[df_rb['Project'] != 'Halo']\n",
    "\n",
    "# drop non AMP AIM analyses\n",
    "df_rb = df_rb[df_rb['Include in AMP AIM analyses?'] != 'NO']\n",
    "\n",
    "# need to replace . with - so it is compatible with fastqs\n",
    "df_rb.index = df_rb.index.map(lambda x: x.replace('.','-'))\n",
    "\n",
    "df_rb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "ef244d1c-a1f1-4414-b54a-80cebbab874d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "267\n",
      "{'134-b-HC-microra', '188-b-HC-microra', '183-b-HC-microra', '99-b-HC-microra', '100-b-HC-microra', 'NEG-4', '186-b-HC-microra', '158-b-HC-microra', '168-b-HC-microra', 'MOC-348', '160-b-HC-microra', 'NEG-5', 'NEG-6', '123-b-HC-microra', '150-b-HC-microra'}\n",
      "{'PA1018-24-dipsa', 'PA1016-00-dipsa', '520-0-twin-psa', '557-0-twin-psa', 'PA1020-00-dipsa', '561905-12-dipsa', 'PA1001-12-dipsa', 'PA1018-12-dipsa', 'PA1004-12-dipsa', '541-0-twin-psa', '544-0-twin-psa', '556-0-twin-psa', 'PA1005-12-dipsa', 'PA1016-24-dipsa', '553-0-twin-psa', '521-0-twin-psa', '532-0-twin-psa', 'PA1020-24-dipsa', '276250-24-dipsa', '528-0-twin-psa', '547-0-twin-psa', '519-0-twin-psa', '571666-12-dipsa', '276250-00-dipsa', 'PA1001-24-dipsa', '552-0-twin-psa', 'PA1002-00-dipsa', 'PA1023-12-dipsa', 'PA1019-00-dipsa', '174086-00-dipsa', '554460-24-dipsa', '561859-24-dipsa', '193246-24-dipsa', '538-0-twin-psa', '525-0-twin-psa', '276250-12-dipsa', '539-0-twin-psa', 'PA1014-00-dipsa', 'PA1023-24-dipsa', '534-0-twin-psa', 'PA1010-24-dipsa', 'PA1014-24-dipsa', '356532-12-dipsa', 'PA1020-12-dipsa', '537-0-twin-psa', '561905-00-dipsa', 'PA1006-24-dipsa', 'PA1008-00-dipsa', '536-0-twin-psa', '571666-24-dipsa', 'PA1004-24-dipsa', '530-0-twin-psa', '356532-00-dipsa', 'PA1014-12-dipsa', 'PA1010-00-dipsa', '555-0-twin-psa', '561905-24-dipsa', 'PA1006-12-dipsa', 'PA1010-12-dipsa', '174086-12-dipsa', 'PA1003-24-dipsa', '200897-12-dipsa', 'PA1001-00-dipsa', 'PA1005-00-dipsa', '531-0-twin-psa', 'PA1005-24-dipsa', 'PA1002-12-dipsa', '560-0-twin-psa', '561859-00-dipsa', 'PA1018-00-dipsa', 'PA1023-00-dipsa', '546-0-twin-psa', '526-0-twin-psa', '540-0-twin-psa', 'PA1006-00-dipsa', '561-0-twin-psa', 'PA1027-12-dipsa', '200897-24-dipsa', 'PA1003-12-dipsa', '535-0-twin-psa', '170148-12-dipsa', 'PA1016-12-dipsa', 'PA1026-00-dipsa', '550-0-twin-psa', '562-0-twin-psa', '563-0-twin-psa', 'PA1002-24-dipsa', 'PA1008-12-dipsa', '518-0-twin-psa', 'PA1008-24-dipsa', '170148-24-dipsa', '529-0-twin-psa', '554460-00-dipsa', '571666-00-dipsa', '543-0-twin-psa', 'PA1004-00-dipsa', '527-2-twin-psa', '542-0-twin-psa', '193246-12-dipsa', '533-0-twin-psa', 'PA1021-00-dipsa', '174086-24-dipsa', 'PA1025-00-dipsa', 'PA1027-00-dipsa', '193246-00-dipsa', '200897-00-dipsa', '356532-24-dipsa', '551-0-twin-psa', 'PA1003-00-dipsa', '170148-00-dipsa', '554-0-twin-psa', '545-0-twin-psa', '561859-12-dipsa', '524-0-twin-psa', '554460-12-dipsa', 'PA1027-24-dipsa'}\n"
     ]
    }
   ],
   "source": [
    "# which samples are missing in our run?\n",
    "rb = list(df_rb.index.values)\n",
    "ac = list(df.index.values)\n",
    "\n",
    "inter = set(rb).intersection(set(ac))\n",
    "print(len(inter))\n",
    "rb_diff = set(rb).difference(set(ac))\n",
    "print(rb_diff)\n",
    "ac_diff = set(ac).difference(set(rb))\n",
    "print(ac_diff)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c609c24-fe31-48b5-8934-20230f4f66bd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
