{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "af7bc565-0ef1-4d28-8bb1-e71b9676b586",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import libraries\n",
    "import pandas as pd\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "\n",
    "# disable warnings, use w caution\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "# project specific libs\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import pathlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b7c9485e-950d-494d-a210-9a5e772dfdbf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# project specific path\n",
    "path = '/Users/KevinBu/Desktop/clemente_lab/Projects/ampaim/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9f389e19-efd3-47df-83c0-7c80b995c261",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['#SampleID', 'LinkerPrimerSequence', 'BarcodeSequence',\n",
       "       'Assay Type', 'AvgSpotLen', 'Bases', 'BioProject', 'BioSample',\n",
       "       'BioSampleModel', 'Bytes', 'Center Name', 'collection_date',\n",
       "       'Consent', 'DATASTORE filetype', 'DATASTORE provider',\n",
       "       'DATASTORE region', 'Experiment', 'geo_loc_name_country',\n",
       "       'geo_loc_name_country_continent', 'geo_loc_name', 'host',\n",
       "       'Instrument', 'isolation_source', 'lat_lon', 'Library Name',\n",
       "       'LibraryLayout', 'LibrarySelection', 'LibrarySource', 'Organism',\n",
       "       'Platform', 'ReleaseDate', 'create_date', 'version', 'Sample Name',\n",
       "       'SRA Study', '1-Methylhistidine', '3-Methylhistidine', 'Alanine',\n",
       "       'allo-Isoleucine', 'alpha-Amino-N-butyric-acid',\n",
       "       'alpha-Aminoadipic-acid', 'Anserine', 'Arginine', 'Asparagine',\n",
       "       'Aspartic_Acid', 'beta-Alanine', 'beta-Aminoisobutyric-acid',\n",
       "       'betaine', 'carnitine', 'Carnosine', 'choline', 'Citrulline',\n",
       "       'Cystathionine_1', 'Cystathionine_2', 'Cystine', 'Ethanolamine',\n",
       "       'gamma-Amino-N-butyric-acid', 'Glutamic_Acid', 'Glutamine',\n",
       "       'Glycine', 'Histidine', 'Homocystine', 'Hydroxylysine_1',\n",
       "       'Hydroxylysine_2', 'Hydroxyproline', 'Isoleucine', 'Leucine',\n",
       "       'Lysine', 'Methionine', 'Ornithine', 'Phenylalanine',\n",
       "       'Phosphoethanolamine', 'proline', 'Sarcosine', 'Serine', 'Taurine',\n",
       "       'Threonine', 'tma', 'TMAO', 'Tryptophan', 'Tyrosine', 'Valine',\n",
       "       'ACPA', 'bmi', 'CRP', 'Eggerthella', 'gender', 'Global', 'HAQ',\n",
       "       'HLADR4', 'Hydroxychloroquine', 'Methotrexate', 'prednisone',\n",
       "       'RadiographicErosions', 'relation', 'rf', 'symptoms'], dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# jobs39 is where the first PRJNA will be tested, 16S sequencing\n",
    "# PRJNA317370\n",
    "# we need to generate a Q2 mapping file like in OA\n",
    "q2_oa = pd.read_csv(path + 'outputs/jobs39/oaq2.tsv', sep='\\t')\n",
    "\n",
    "# take columns i think we need\n",
    "keep = ['BarcodeSequence','LinkerPrimerSequence']\n",
    "\n",
    "# grab sample IDs\n",
    "# df_map = q2_oa.loc[:,keep]\n",
    "\n",
    "df_sra = pd.read_csv(path + 'outputs/jobs39/SraRunTable.txt', sep='\\t')\n",
    "df_sra = df_sra.rename(columns={'Run':'#SampleID'})\n",
    "\n",
    "# insert in reverse order\n",
    "for i in keep:\n",
    "    df_sra.insert(1, i, ['NA' for _ in range(len(df_sra))])\n",
    "\n",
    "df_sra.to_csv(path + 'outputs/jobs39/qiime_mapping_file.tsv', sep='\\t')\n",
    "df_sra.columns.values\n",
    "# df_sra['Sample Name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "53b8d802-bc26-4d98-b54d-07864b8037b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47.233333333333334 7.103584978039602\n",
      "63.05 5.780866808068802\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([24, 36, 37, 40, 41, 42, 42, 44, 44, 45, 46, 47, 48, 48, 48, 48, 48,\n",
       "       48, 49, 50, 52, 52, 53, 54, 54, 55, 55, 55, 56, 56, 56, 57, 57, 57,\n",
       "       57, 59, 59, 60, 60, 61, 63, 64, 65, 65, 66, 67, 68, 71, 74, 75])"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# jobs44\n",
    "# one partition of HC n=30 has mean age 55.5 (8.80)\t\n",
    "# other partition of 20 disease has mean age 50.7 (10.1)\n",
    "df_sra = pd.read_csv(path + 'outputs/jobs44/SraRunTable.csv')\n",
    "df = df_sra[['age','sex']]\n",
    "df = df.sort_values(by='age',ascending=True)\n",
    "\n",
    "df_HC = df.iloc[0:30,:]\n",
    "df_SjD = df.iloc[30:,:]\n",
    "\n",
    "for df_ in [df_HC, df_SjD]:\n",
    "    print(np.mean(df_.age),np.std(df_.age,ddof=1))\n",
    "\n",
    "df.age.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "cb773109-1ed0-4b67-a9dc-633ce280e477",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Run', 'age', 'Assay Type', 'AvgSpotLen', 'Bases',\n",
       "       'BIOMATERIAL_PROVIDER', 'BioProject', 'BioSample',\n",
       "       'BioSampleModel', 'Bytes', 'Center Name', 'collection_data',\n",
       "       'Consent', 'DATASTORE filetype', 'DATASTORE provider',\n",
       "       'DATASTORE region', 'Experiment', 'Instrument', 'isolate',\n",
       "       'Library Name', 'LibraryLayout', 'LibrarySelection',\n",
       "       'LibrarySource', 'Organism', 'Platform', 'ReleaseDate',\n",
       "       'create_date', 'version', 'Sample Name', 'sex', 'SRA Study',\n",
       "       'tissue'], dtype=object)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_sra.columns.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "00efecc2-e350-4c6d-9757-c79ca6cbb524",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "found\n",
      "[36 42 41 52 48 48 37 75 46 56 57 52 24 56 55 68 59 64 50 48]\n",
      "50.7 11.662355093024285\n",
      "[63 42 60 53 57 61 65 74 48 48 67 55 56 57 57 60 44 54 40 49 45 71 54 59\n",
      " 48 47 44 55 65 66]\n",
      "55.46666666666667 8.795505644374852\n",
      "found\n",
      "[60 46 24 59 55 41 42 66 52 75 57 59 37 47 40 56 44 42 60 50]\n",
      "50.6 11.618497049283548\n",
      "[54 55 67 56 57 65 44 48 64 49 52 53 57 65 48 48 54 55 48 45 56 48 57 68\n",
      " 74 61 71 63 36 48]\n",
      "55.53333333333333 8.795505644374852\n",
      "found\n",
      "[56 71 64 61 49 42 63 56 52 67 37 40 46 24 48 45 36 53 55 48]\n",
      "50.65 11.63603748795322\n",
      "[42 41 47 75 66 48 57 57 59 52 68 48 44 55 44 54 60 65 56 57 60 48 55 57\n",
      " 48 65 50 59 74 54]\n",
      "55.5 8.799490580866522\n",
      "found\n",
      "[45 63 61 56 42 48 48 36 65 55 37 48 64 41 24 60 67 59 54 40]\n",
      "50.65 11.63603748795322\n",
      "[48 50 52 53 48 56 47 75 59 42 68 44 56 74 55 57 57 57 46 54 48 44 55 71\n",
      " 57 60 52 49 65 66]\n",
      "55.5 8.799490580866522\n",
      "found\n",
      "[55 48 46 68 71 36 48 57 42 47 48 60 37 55 44 45 24 56 60 67]\n",
      "50.7 11.657841273490336\n",
      "[53 61 42 49 41 52 66 59 44 75 57 63 57 55 54 57 65 74 48 59 56 64 48 56\n",
      " 50 40 54 52 65 48]\n",
      "55.46666666666667 8.799425268588356\n",
      "found\n",
      "[49 44 45 37 53 59 74 57 66 41 36 50 24 55 48 56 65 42 55 57]\n",
      "50.65 11.640559762268627\n",
      "[54 48 52 55 71 56 42 48 47 61 57 54 44 56 48 65 57 75 67 48 64 63 59 40\n",
      " 68 52 60 60 46 48]\n",
      "55.5 8.795570985758527\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "def random_split_array(arr, split_ratio):\n",
    "    np.random.shuffle(arr)  # Shuffle the array in place\n",
    "    split_index = int(len(arr) * split_ratio)\n",
    "    arr1 = arr[:split_index]\n",
    "    arr2 = arr[split_index:]\n",
    "    return arr1, arr2\n",
    "\n",
    "# grab all combinations of ages\n",
    "# combos = list(itertools.combinations(df.age, 20))\n",
    "\n",
    "# Example usage:\n",
    "my_array = df.age.values\n",
    "\n",
    "\n",
    "for i in range(100000):\n",
    "    split_ratio = 0.4\n",
    "    arr1, arr2 = random_split_array(my_array, split_ratio)\n",
    "    \n",
    "    mu1 = np.mean(arr1)\n",
    "    s1 = np.std(arr1,ddof=1)\n",
    "    \n",
    "    mu2 = np.mean(arr2)\n",
    "    s2 = np.std(arr2,ddof=1)\n",
    "    \n",
    "    if 55.44 < mu2 and mu2 < 55.54:\n",
    "        if 8.794 < s2 and s2 < 8.805:\n",
    "            #if 50.54 < mu1 and mu1 < 50.74:\n",
    "             #   if 10.0 < s1 and s1 < 10.2:\n",
    "            print('found')\n",
    "            print(arr1)\n",
    "            print(mu1,s1)\n",
    "            print(arr2)\n",
    "            print(mu2,s2)\n",
    "\n",
    "#np.testing.assert_approx_equal(mu1, 50.7, significant=2)\n",
    "#np.testing.assert_approx_equal(s1, 10.1, significant=2)\n",
    "#np.testing.assert_approx_equal(mu2, 55.5, significant=2)\n",
    "#np.testing.assert_approx_equal(s2, 8.80, significant=3)\n",
    "\n",
    "print('Done')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "94223283-4bc5-455f-ab6b-8c62af5ba587",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "label\n",
       "A    30\n",
       "B    20\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.label.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f1c4d42b-7277-4f5e-aa77-b3bc804209a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Run title</th>\n",
       "      <th>BioProject accession</th>\n",
       "      <th>Experiment accession</th>\n",
       "      <th>Run data file type</th>\n",
       "      <th>Read filename 1</th>\n",
       "      <th>Read file1 MD5</th>\n",
       "      <th>DownLoad Read file1</th>\n",
       "      <th>Read filename 2</th>\n",
       "      <th>Read file2 MD5</th>\n",
       "      <th>...</th>\n",
       "      <th>DownLoad Index file2</th>\n",
       "      <th>Reference file name</th>\n",
       "      <th>MD5 for reference file</th>\n",
       "      <th>Assembly Name or Accession</th>\n",
       "      <th>Assembly Accession URL</th>\n",
       "      <th>other_db</th>\n",
       "      <th>accession_in_other_db</th>\n",
       "      <th>other_db_url</th>\n",
       "      <th>Diagnosis</th>\n",
       "      <th>HostSubjectId</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>#q2:types</th>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>...</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "      <td>categorical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRR442594</th>\n",
       "      <td>2</td>\n",
       "      <td>D2:Amplicon sequence of human fecal sample:non...</td>\n",
       "      <td>PRJCA008752</td>\n",
       "      <td>CRX385340</td>\n",
       "      <td>fastq</td>\n",
       "      <td>CRR442594_f1.fq.gz (12307086 bytes)</td>\n",
       "      <td>b7f3cf4ef58a2909b282c04ea9ca4291</td>\n",
       "      <td>ftp://download.big.ac.cn/gsa/CRA006415/CRR4425...</td>\n",
       "      <td>CRR442594_r2.fq.gz (11537276 bytes)</td>\n",
       "      <td>2f8c75231ad1ca51500d811f4aaa5124</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-pSS patient</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRR442598</th>\n",
       "      <td>6</td>\n",
       "      <td>D6:Amplicon sequence of human fecal sample:non...</td>\n",
       "      <td>PRJCA008752</td>\n",
       "      <td>CRX385344</td>\n",
       "      <td>fastq</td>\n",
       "      <td>CRR442598_f1.fq.gz (10916305 bytes)</td>\n",
       "      <td>c4a7ee34c4b46ef345cf873e5cf2bf00</td>\n",
       "      <td>ftp://download.big.ac.cn/gsa/CRA006415/CRR4425...</td>\n",
       "      <td>CRR442598_r2.fq.gz (10506223 bytes)</td>\n",
       "      <td>9efcb1b6596069129c67c2a87531d40e</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non-pSS patient</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRR442600</th>\n",
       "      <td>8</td>\n",
       "      <td>D8:Amplicon sequence of human fecal sample:pSS...</td>\n",
       "      <td>PRJCA008752</td>\n",
       "      <td>CRX385346</td>\n",
       "      <td>fastq</td>\n",
       "      <td>CRR442600_f1.fq.gz (12975507 bytes)</td>\n",
       "      <td>3149e0d5588d0cea4b58c1a2706c7e43</td>\n",
       "      <td>ftp://download.big.ac.cn/gsa/CRA006415/CRR4426...</td>\n",
       "      <td>CRR442600_r2.fq.gz (12152647 bytes)</td>\n",
       "      <td>038b8b7280f629bba6e6fa50eb5b78f2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pSS patients without treament</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRR442603</th>\n",
       "      <td>11</td>\n",
       "      <td>D11:Amplicon sequence of human fecal sample:pS...</td>\n",
       "      <td>PRJCA008752</td>\n",
       "      <td>CRX385349</td>\n",
       "      <td>fastq</td>\n",
       "      <td>CRR442603_f1.fq.gz (10712320 bytes)</td>\n",
       "      <td>854db81c703df4130019c80e717ccad7</td>\n",
       "      <td>ftp://download.big.ac.cn/gsa/CRA006415/CRR4426...</td>\n",
       "      <td>CRR442603_r2.fq.gz (10213484 bytes)</td>\n",
       "      <td>692bcab19cb7f412000af2869de5355f</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pSS patients without treament</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID                                          Run title  \\\n",
       "#SampleID                                                                   \n",
       "#q2:types  categorical                                        categorical   \n",
       "CRR442594            2  D2:Amplicon sequence of human fecal sample:non...   \n",
       "CRR442598            6  D6:Amplicon sequence of human fecal sample:non...   \n",
       "CRR442600            8  D8:Amplicon sequence of human fecal sample:pSS...   \n",
       "CRR442603           11  D11:Amplicon sequence of human fecal sample:pS...   \n",
       "\n",
       "          BioProject accession Experiment accession Run data file type  \\\n",
       "#SampleID                                                                \n",
       "#q2:types          categorical          categorical        categorical   \n",
       "CRR442594          PRJCA008752            CRX385340              fastq   \n",
       "CRR442598          PRJCA008752            CRX385344              fastq   \n",
       "CRR442600          PRJCA008752            CRX385346              fastq   \n",
       "CRR442603          PRJCA008752            CRX385349              fastq   \n",
       "\n",
       "                               Read filename 1  \\\n",
       "#SampleID                                        \n",
       "#q2:types                          categorical   \n",
       "CRR442594  CRR442594_f1.fq.gz (12307086 bytes)   \n",
       "CRR442598  CRR442598_f1.fq.gz (10916305 bytes)   \n",
       "CRR442600  CRR442600_f1.fq.gz (12975507 bytes)   \n",
       "CRR442603  CRR442603_f1.fq.gz (10712320 bytes)   \n",
       "\n",
       "                             Read file1 MD5  \\\n",
       "#SampleID                                     \n",
       "#q2:types                       categorical   \n",
       "CRR442594  b7f3cf4ef58a2909b282c04ea9ca4291   \n",
       "CRR442598  c4a7ee34c4b46ef345cf873e5cf2bf00   \n",
       "CRR442600  3149e0d5588d0cea4b58c1a2706c7e43   \n",
       "CRR442603  854db81c703df4130019c80e717ccad7   \n",
       "\n",
       "                                         DownLoad Read file1  \\\n",
       "#SampleID                                                      \n",
       "#q2:types                                        categorical   \n",
       "CRR442594  ftp://download.big.ac.cn/gsa/CRA006415/CRR4425...   \n",
       "CRR442598  ftp://download.big.ac.cn/gsa/CRA006415/CRR4425...   \n",
       "CRR442600  ftp://download.big.ac.cn/gsa/CRA006415/CRR4426...   \n",
       "CRR442603  ftp://download.big.ac.cn/gsa/CRA006415/CRR4426...   \n",
       "\n",
       "                               Read filename 2  \\\n",
       "#SampleID                                        \n",
       "#q2:types                          categorical   \n",
       "CRR442594  CRR442594_r2.fq.gz (11537276 bytes)   \n",
       "CRR442598  CRR442598_r2.fq.gz (10506223 bytes)   \n",
       "CRR442600  CRR442600_r2.fq.gz (12152647 bytes)   \n",
       "CRR442603  CRR442603_r2.fq.gz (10213484 bytes)   \n",
       "\n",
       "                             Read file2 MD5  ... DownLoad Index file2  \\\n",
       "#SampleID                                    ...                        \n",
       "#q2:types                       categorical  ...          categorical   \n",
       "CRR442594  2f8c75231ad1ca51500d811f4aaa5124  ...                  NaN   \n",
       "CRR442598  9efcb1b6596069129c67c2a87531d40e  ...                  NaN   \n",
       "CRR442600  038b8b7280f629bba6e6fa50eb5b78f2  ...                  NaN   \n",
       "CRR442603  692bcab19cb7f412000af2869de5355f  ...                  NaN   \n",
       "\n",
       "          Reference file name MD5 for reference file  \\\n",
       "#SampleID                                              \n",
       "#q2:types         categorical            categorical   \n",
       "CRR442594                 NaN                    NaN   \n",
       "CRR442598                 NaN                    NaN   \n",
       "CRR442600                 NaN                    NaN   \n",
       "CRR442603                 NaN                    NaN   \n",
       "\n",
       "          Assembly Name or Accession Assembly Accession URL     other_db  \\\n",
       "#SampleID                                                                  \n",
       "#q2:types                categorical            categorical  categorical   \n",
       "CRR442594                        NaN                    NaN          NaN   \n",
       "CRR442598                        NaN                    NaN          NaN   \n",
       "CRR442600                        NaN                    NaN          NaN   \n",
       "CRR442603                        NaN                    NaN          NaN   \n",
       "\n",
       "          accession_in_other_db other_db_url                      Diagnosis  \\\n",
       "#SampleID                                                                     \n",
       "#q2:types           categorical  categorical                    categorical   \n",
       "CRR442594                   NaN          NaN                non-pSS patient   \n",
       "CRR442598                   NaN          NaN                non-pSS patient   \n",
       "CRR442600                   NaN          NaN  pSS patients without treament   \n",
       "CRR442603                   NaN          NaN  pSS patients without treament   \n",
       "\n",
       "          HostSubjectId  \n",
       "#SampleID                \n",
       "#q2:types   categorical  \n",
       "CRR442594             2  \n",
       "CRR442598             6  \n",
       "CRR442600             8  \n",
       "CRR442603            11  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create mapping file\n",
    "df = pd.read_csv(path + 'outputs/jobs45/gutonlyCRR.csv')\n",
    "df = df[df['Run title'].str.contains('fecal')]\n",
    "df = df.set_index('Accession')\n",
    "\n",
    "# get diagnosis column\n",
    "df['Diagnosis'] = df['Run title'].map(lambda x: x.split('sample:')[-1])\n",
    "\n",
    "# create categorical\n",
    "df = df.reset_index()\n",
    "df.loc[-1] = ['categorical' for _ in range(len(df.columns))] # adding a row\n",
    "df.index = df.index + 1  # shifting index\n",
    "df.sort_index(inplace=True) \n",
    "df.iloc[0,0] = '#q2:types'\n",
    "df = df.set_index('Accession')\n",
    "df.index.name = '#SampleID'\n",
    "\n",
    "# create host subject id\n",
    "df['HostSubjectId'] = df['ID']\n",
    "\n",
    "# export\n",
    "df.to_csv(path + 'outputs/jobs45/qiime_mapping_file.tsv', sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c2d4c87-be9d-427f-a839-6ed1bfa65407",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
